framework/test-vectors/test-vector-enumeration.md

```diff
@@ -0,0 +1,268 @@
+
+For example, we might partition the set of all possible plaintexts
+based on the plaintext length.
+Our hypothetical "smaller sets" might be:
+
+- Empty: length = 0
+- Small: all plaintexts where (1 < length ≤ 10)
+- Medium: all plaintexts where (10 < length ≤ 1000)
+- Large: all plaintexts where (1000 < length ≤ 2^32-1)
+- Largest frame: all plaintexts where (length = 2^32-1)
+- Largest frame + partial frame: all plaintexts where (length = 2^32-1 + [1 .. 2^32-1))
+- Two largest frames: all plaintexts where (length = 2\*(2^32-1))
+- Many frames: all plaintexts where (length = 2*(2^32-1) + [1 .. (2^32-1)*(2^32-3)])
+
+(The motivation for partitioning this set based on lengths of 2^32-1 comes from the ESDK message [frame size](https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/message-format.html#body-framing).)
+
+From each of these smaller sets, we select one (or more)
+concrete "representative" values that are tested in test vectors.
+For example, a representative "small" plaintext could be `"abc"`.
+
+This framework of representative values
+reduces any impractically large set of test inputs
+to a practically testable size
+while maintaining reasonable test coverage.
+Its specification here documents reasoning for this framework
+for readers or users of a test vector library
+to understand the motivation for this concept.
+
+### Modifying representative values
+
+It is expected that a representative value could be modified
+within its constraints
+without loss in test coverage.
+
+For example,
+given a representative small plaintext of `"abc"`,
+if one were to change this to `"wxyz"`,
+this should have no impact on test coverage.
+This is still a small plaintext
+according to the constraints on small plaintexts
+and should be an allowed change.
+
+On the other hand,
+given a representative small plaintext of `"abc"`,
+if one were to change this to `""`,
+this removes coverage for small plaintexts.
+This violates the constraint for small plaintexts
+and should not be an allowed change.
+
+The property that a representative value may be modified within its constraints
+acts as an assertion that the representative value is not special,
+and is truly "representative" of other values in its constrained set.
+
+### Filtering input configurations
+
+Input configurations may be "filtered"
+and not be tested.
+
+This is a mechanism to identify test cases that
+may not be as important as others
+and avoid writing them to a test vector manifest.
+
+#### Motivation
+
+Some input configurations produced as a result of the enumeration above
+will not be tested
+as they are not important enough
+to warrant the time spent testing them.
+
+For example,
+consider all cases where a hierarchical keyring is encrypting,
+but some raw AES key is decrypting.
+This is expected have an [expected result](#expected-results) of `negative-decrypt`,
+since a hierarchical keyring's encrypt result is incompatible with AES decryption.
```

Comment:
Nit/Suggestion: AES is Advanced Encryption Standard, which is an Algorithm, and is used by H-Keyring.
```suggestion
since a hierarchical keyring's encrypt result is incompatible with a AES Keyring's decryption.
```